File: 1_Two_Sum/xsy/solution.go
package _1

// twoSum遍历
func twoSum(nums []int, target int) []int {
    for i := 0; i<len(nums)-1; i++ {
        for j := i+1; j<len(nums); j++ {
            if nums[i] + nums[j] == target {
                return []int{i, j}
            }
        }
    }
    return []int{}   
}

// twoSum2使用map存储
func twoSum2(nums []int, target int) []int {
    m := map[int]int{}
    for i, v := range nums {
        if j, ok := m[v]; ok {
            return []int{j, i}
        } else {
            m[target-v] = i
        }
    }
    return []int{}
}

File: 1_Two_Sum/xsy/solution.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = dict()
        for index, num in enumerate(nums):
            if num in m:
                return [m[num], index]
            else:
                m[target-num] = index
        return []
